### PR TITLE
Added some documentation to class ChunkFiller, fixed a bug in getFreeRndPos...

### PR DESCRIPTION
**Pull request by grauerkoala**
… where the returned position could be outside the chunk.

The documentation is, of course, by no means complete, but it's a start.

### CODE REVIEW

File: main/src/com/miloshpetrov/sol2/game/chunk/ChunkFiller.java [hunk 1]
```diff
@@ -163,83 +172,162 @@ private void fillAsteroids(SolGame game, RemoveController remover, boolean forBe
...
+   * @param g        The {@link SolGame} instance to work with
+   * @param chCenter The center of a chunk in which a random position should be found
+   * @return A random, unoccupied position in a chunk centered around chCenter, relative to the entire map, or <code>null</code> if within 100 tries no unoccupied position has been found
+   */
   private Vector2 getFreeRndPos(SolGame g, Vector2 chCenter) {
-    Vector2 pos = new Vector2(chCenter);
     for (int i = 0; i < 100; i++) {
-      pos.x += SolMath.rnd(Const.CHUNK_SIZE/2);
-      pos.y += SolMath.rnd(Const.CHUNK_SIZE/2);
+      Vector2 pos = getRndPos(new Vector2(chCenter));
```
Comment (SimonC4) [inline main/src/com/miloshpetrov/sol2/game/chunk/ChunkFiller.java]: Does this change work as expected? Previously it was adding a random amount to x and y positions every iteration in the for loop. After the change you are getting a new vector position every iteration.

Comment (grauerkoala) [inline main/src/com/miloshpetrov/sol2/game/chunk/ChunkFiller.java]: This works differently, but on purpose. The way the method was originally designed, a position outside of the chunk whose center chCenter represents could be chosen; for example, if the first value hits exactly (-1)*Const.CHUNK_SIZE/2, but an object is already occupying that spot, a new random value is going to be added on top of that. If that value is negative again, the position will end up being outside the chunk. I'm fairly confident that this is unintended behaviour, which is why I changed it in such a manner that it now always chooses a random position based on the center of the chunk and not some random position therein.

Comment (SimonC4) [inline main/src/com/miloshpetrov/sol2/game/chunk/ChunkFiller.java]: Ok, sounds good and I'm glad it was intentional :+1:

Comment (grauerkoala) [inline main/src/com/miloshpetrov/sol2/game/chunk/ChunkFiller.java]: I'm glad you're keeping an eye out nonetheless. :)

**Comment by Cervator** [comment]
Tested around a bit, seems we still have plenty of junk, dust, and debris! Merged :-)

**Pull request merged**